### PR TITLE
Create work packages on IFC model page

### PR DESCRIPTION
**Pull request by ulferts**
Adds the capability to create work_packages from the ifc_models page. The L&F is similar to how work packages are created on the work package list page.

## TODOs

* [x] Check permissions to disable the button if the `add_work_packages` permission is lacking.
* [x] ~~Add the ability to add BCFs to the `ifc_models/:id` page as well.~~ No longer necessary since route refactoring
* [x] Add the split create on the `list-only` mode as well
* [x] Generalize the split screen create for work package and bim using the route template
* [x] Allow creating subsequent BCFs. Currently, the button is disabled.
* [x] Styling so that e.g. the lower right save bar does not have a bottom margin. 

## Out of scope

* Add a viewpoint to the created work package
  * Error handling (transaction) when one of the requests fail. We need to remove the work package again in such cases but might not be allowed to.
* Decide on additional places where BCFs are to be created. 
*  Different success messages upon creation. Currently, a "The work package has been successfully created" message is displayed.

https://community.openproject.com/projects/openproject/work_packages/30519

### CODE REVIEW

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/150941977b158758cd4d8027ba77f28223d2fe25/frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L37">frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: frontend/src/app/components/wp-new/wp-new-full-view.component.ts [hunk 1]
```diff
@@ -26,14 +26,14 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {WorkPackageCreateController} from 'core-components/wp-new/wp-create.controller';
+import {WorkPackageCreateComponent} from 'core-components/wp-new/wp-create.component';
 import {Component} from '@angular/core';
```
Comment (openprojectci) [inline frontend/src/app/components/wp-new/wp-new-full-view.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/8bc9e1f18f1494c2afdb99f865cf858298ae755e/frontend/src/app/components/wp-new/wp-new-full-view.component.ts#L31"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-new/wp-new-split-view.component.ts [hunk 2]
```diff
@@ -26,13 +26,13 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {WorkPackageCreateController} from 'core-components/wp-new/wp-create.controller';
+import {WorkPackageCreateComponent} from 'core-components/wp-new/wp-create.component';
 import {Component} from '@angular/core';
```
Comment (openprojectci) [inline frontend/src/app/components/wp-new/wp-new-split-view.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/8bc9e1f18f1494c2afdb99f865cf858298ae755e/frontend/src/app/components/wp-new/wp-new-split-view.component.ts#L31"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/081ba59455a4ea64fa1f7c7f1e0a8120f02c3ee4/frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L45">frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L45</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by HDinger — CHANGES_REQUESTED** [review]
All in one it works. But I found some issues:

* When the details split screen is open, a click on the create button does nothing (on the ifc page).
* When starting on the ifc defaults page, there is a border shown around the card view and the spacing to the right is missing. Both will vanish, when going to the details view of a WP and back again to the card view. I assume that a class is not correctly set.
Bildschirmfoto 2020-03-04 um 10 16 05

* A similar behaviour happens on the WP page: After opening the split screen and closing it again, there is too much spacing to the right:
Bildschirmfoto 2020-03-04 um 10 14 48

* When starting on `/work_packages` and opening the create form, the button is not disabled:
Bildschirmfoto 2020-03-04 um 10 15 08
If you then reload the page and close the create form, the button stays disabled:
Bildschirmfoto 2020-03-04 um 10 15 23
The same happens on the ifc page.

**Comment by oliverguenther** [comment]
@HDinger I think I caught all of them

**Review by HDinger — COMMENTED** [review]
@oliverguenther One last thing: When I am on `projects/foo/bcf/list?models...` and click on create, the state changes to the viewer and list. I guess it should stay in the list only with an opened create form.

**Review by HDinger — APPROVED** [review]
👍 Good to merge once travis is ready

File: frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ import {Component, Injector, OnInit} from '@angular/core';
 import {WorkPackageViewSelectionService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-selection.service';
 import {WorkPackageSingleViewBase} from "core-app/modules/work_packages/routing/wp-view-base/work-package-single-view.base";
 import {BackRoutingService} from "core-app/modules/common/back-routing/back-routing.service";
+import {of} from "rxjs";
```
Comment (openprojectci) [inline frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts:38]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/081ba59455a4ea64fa1f7c7f1e0a8120f02c3ee4/frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Pull request merged**